src/main/scala/ru/d4team/chat/db/DBMigrator.scala

```diff
@@ -0,0 +1,45 @@
+package ru.d4team.chat.db
+
+import org.flywaydb.core.api.MigrationState
+import org.flywaydb.core.api.configuration.FluentConfiguration
+import zio._
+
+
+object DBMigrator {
+
+  def migrate: ZIO[FluentConfiguration, Throwable, Unit] =
+    for {
+      flywayConfig <- ZIO.service[FluentConfiguration]
+      _            <- ZIO.logInfo(s"Starting the migration for host: ${flywayConfig.getUrl}")
+      _            <- logValidationErrorsIfAny(flywayConfig)
+      _            <- ZIO.logInfo("Migrations validation successful")
+      count        <- succeedMigrationCount(flywayConfig)
+      _            <- checkStatesForSuccess(flywayConfig)
+      _            <- ZIO.logInfo(s"Successful migrations: $count")
+    } yield ()
+
+  private def succeedMigrationCount(config: FluentConfiguration): IO[Throwable, Int] = {
+    ZIO.succeed(config.load().migrate().migrationsExecuted)
+  }
+
+  private def checkStatesForSuccess(config: FluentConfiguration): IO[Error, List[Unit]] = {
+    val infoWithStates = config.load().info().all().toList.map(info => (info, info.getState))
+    ZIO.foreach(infoWithStates) {
+      case (_, MigrationState.SUCCESS) => ZIO.unit
+      case (info, e) => ZIO.fail(new Error(s"Migration ${info.getDescription} status is not \"SUCCESS\": ${e.toString}"))
+    }
+  }
+  private def logValidationErrorsIfAny(flywayConfig: FluentConfiguration): ZIO[Any, Throwable, Unit] =
+    for {
+      validated <- ZIO.succeed(
+        flywayConfig
+          .ignoreMigrationPatterns("*:pending")
+          .load()
+          .validateWithResult
+      )
+      _ <- ZIO.when(!validated.validationSuccessful)(
+        ZIO.fail(new Error(validated.getAllErrorMessages + "\n Migrations validation failed"))
+      )
+    } yield ()
```

Comment:
It might be simplified:
```scala
private def logValidationErrorsIfAny(conf: FluentConfiguration): ZIO[Any, Throwable, ValidateResult] =
  ZIO
    .attemptBlocking(conf.ignoreMigrationPatterns("*:pending").load.validateWithResult)
    .filterOrElseWith(_.validationSuccessful) { validateResult =>
      ZIO.fail(new Error(s"Migrations validation failed:\n${validateResult.getAllErrorMessages}"))
    }
```

Reply:
thank you for this simplification) Done